### ISSUE TITLE
Implement bank-account properly

### ISSUE THREAD

**Issue by yurrriq** [exercism/xclojure#67]
```bash
$ cd bank-account
$ ln -s example.clj bank_account.clj # cheap hack
$ lein test
```

```
Testing bank-account-test

Ran 4 tests containing 8 assertions.
0 failures, 0 errors.

lein test bank-account

lein test bank-account-test

lein test :only bank-account-test/check-concurrent-access

ERROR in (check-concurrent-access) (ThreadPoolExecutor.java:2047)
Uncaught exception, not in assertion.
expected: nil
  actual: java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.FutureTask@8ad6665 rejected from java.util.concurrent.ThreadPoolExecutor@30af5b6b[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 3]
 at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution (ThreadPoolExecutor.java:2047)
    java.util.concurrent.ThreadPoolExecutor.reject (ThreadPoolExecutor.java:823)
    java.util.concurrent.ThreadPoolExecutor.execute (ThreadPoolExecutor.java:1369)
    java.util.concurrent.AbstractExecutorService.submit (AbstractExecutorService.java:134)
    clojure.core$future_call.invoke (core.clj:6680)
    clojure.core$pmap$fn__6744.invoke (core.clj:6729)
    clojure.core$map$fn__4553.invoke (core.clj:2624)
    clojure.lang.LazySeq.sval (LazySeq.java:40)
    clojure.lang.LazySeq.seq (LazySeq.java:49)
    clojure.lang.RT.seq (RT.java:507)
    clojure.lang.RT.nthFrom (RT.java:924)
    clojure.lang.RT.nth (RT.java:883)
    clojure.core$pmap$step__6749.invoke (core.clj:6730)
    clojure.core$pmap.invoke (core.clj:6735)
    clojure.core$pcalls.doInvoke (core.clj:6749)
    clojure.lang.RestFn.invoke (RestFn.java:436)
    bank_account_test/fn (bank_account_test.clj:51)
    clojure.test$test_var$fn__7670.invoke (test.clj:704)
    clojure.test$test_var.invoke (test.clj:704)
    clojure.test$test_vars$fn__7692$fn__7697.invoke (test.clj:722)
    clojure.test$default_fixture.invoke (test.clj:674)
    clojure.test$test_vars$fn__7692.invoke (test.clj:722)
    clojure.test$default_fixture.invoke (test.clj:674)
    clojure.test$test_vars.invoke (test.clj:718)
    clojure.test$test_all_vars.invoke (test.clj:728)
    clojure.test$test_ns.invoke (test.clj:747)
    clojure.core$map$fn__4553.invoke (core.clj:2624)
    clojure.lang.LazySeq.sval (LazySeq.java:40)
    clojure.lang.LazySeq.seq (LazySeq.java:49)
    clojure.lang.Cons.next (Cons.java:39)
    clojure.lang.RT.boundedLength (RT.java:1735)
    clojure.lang.RestFn.applyTo (RestFn.java:130)
    clojure.core$apply.invoke (core.clj:632)
    clojure.test$run_tests.doInvoke (test.clj:762)
    clojure.lang.RestFn.applyTo (RestFn.java:137)
    clojure.core$apply.invoke (core.clj:630)
    user$eval85$fn__140$fn__171.invoke (form-init7525173329464848704.clj:1)
    user$eval85$fn__140$fn__141.invoke (form-init7525173329464848704.clj:1)
    user$eval85$fn__140.invoke (form-init7525173329464848704.clj:1)
    user$eval85.invoke (form-init7525173329464848704.clj:1)
    clojure.lang.Compiler.eval (Compiler.java:6782)
    clojure.lang.Compiler.eval (Compiler.java:6772)
    clojure.lang.Compiler.load (Compiler.java:7227)
    clojure.lang.Compiler.loadFile (Compiler.java:7165)
    clojure.main$load_script.invoke (main.clj:275)
    clojure.main$init_opt.invoke (main.clj:280)
    clojure.main$initialize.invoke (main.clj:308)
    clojure.main$null_opt.invoke (main.clj:343)
    clojure.main$main.doInvoke (main.clj:421)
    clojure.lang.RestFn.invoke (RestFn.java:421)
    clojure.lang.Var.invoke (Var.java:383)
    clojure.lang.AFn.applyToHelper (AFn.java:156)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.main.main (main.java:37)

Ran 4 tests containing 8 assertions.
0 failures, 1 errors.
Tests failed.
```

**Comment by yurrriq** [comment]
See broken implementation (now removed) [here][1].

[1]: https://github.com/exercism/xclojure/tree/86ebb5bca80e7d3ea091c1bb7113736246bc02fd/bank-account

**Comment by haus** [comment]
@yurrriq are you sure the implementation is broken? I found simply moving the `(shutdown-agents)` call into the bank-test macro allows the tests to pass.

**Comment by haus** [comment]
(deleted my comment after reading more about shutdown-agents)

**Comment by yurrriq** [comment]
Fixed inn #141
